From e76e20d4d4a39e139a778411a2ed09ec35c2046e Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 19 Jul 2011 09:29:28 +1000
Subject: Dont call xf86DeleteInput on PreInit failure.

The server calls this for us once PreInit returns with an error,
calling it during PreInit means the server continues to use
already free'd memory.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

diff --git a/src/acecad.c b/src/acecad.c
index 2b6aa45..a3326a0 100644
--- src/acecad.c
+++ src/acecad.c
@@ -471,7 +471,6 @@ SetupProc_fail:
 	if (local)
 		local->private = NULL;
     }
-    xf86DeleteInput(local, 0);
     return BadAlloc;
 }
 
-- 
cgit v0.10.2

